inc/html.class.php

```diff
@@ -2740,118 +2750,90 @@ static function showMassiveActions($options = []) {
-                  showOtherMonths: true,
-                  selectOtherMonths: true,
-                  showButtonPanel: true,
-                  changeMonth: true,
-                  changeYear: true,
-                  showOn: 'both',
-                  showWeek: true,
-                  buttonText: '<i class=\'far fa-calendar-alt\'></i>'";
-
-      if (!$p['canedit']) {
-         $js .= ",disabled: true";
-      }
-
-      if (!empty($p['min'])) {
-         $js .= ",minDate: '".self::convDate($p['min'])."'";
-      }
-
-      if (!empty($p['max'])) {
-         $js .= ",maxDate: '".self::convDate($p['max'])."'";
-      }
-
-      if (!empty($p['yearrange'])) {
-         $js .= ",yearRange: '". $p['yearrange'] ."'";
-      }
-
-      switch ($_SESSION['glpidate_format']) {
-         case 1 :
-            $p['showyear'] ? $format='dd-mm-yy' : $format='dd-mm';
-            break;
-
-         case 2 :
-            $p['showyear'] ? $format='mm-dd-yy' : $format='mm-dd';
-            break;
-
-         default :
-            $p['showyear'] ? $format='yy-mm-dd' : $format='mm-dd';
-      }
-      $js .= ",dateFormat: '".$format."'";
-
-      if ($p['multiple']) {
-         // Fix altField date format
-         // onSelect callback will be called by multiDatePicker after update of the altField
-         $js .= ",onSelect: function(date, datepicker) {
-               normalizeMultiDateAltField($('#hiddendate{$p['rand']}'), '{$format}');
-            }";
-      }
-
-      $js .= "}).next('.ui-datepicker-trigger').addClass('pointer');";
-
-      if ($p['multiple']) {
-         // Fix altField date format that has just been filled by multiDatePicker
-         $js .= "normalizeMultiDateAltField($('#hiddendate{$p['rand']}'), '{$format}');";
-      }
+      $values = true === $p['multiple']
+         ? explode(', ', $p['value'])
+         : [$p['value']];
+      $values = implode(', ', array_map('Html::convDate', $values));
+
+      $required = $p['required'] == true
+         ? " required='required'"
+         : "";
+      $disabled = !$p['canedit']
+         ? " disabled='disabled'"
+         : "";
+      $clear    = $p['maybeempty'] && $p['canedit']
+         ? "<a data-clear  title='".__s('Clear')."'>
+               <i class='fa fa-times-circle pointer'></i>
+            </a>"
+         : "";
+
+      $output = <<<HTML
+      <div class="no-wrap flatpickr" id="showdate{$p['rand']}">
+         <input type="text" name="{$name}" value="{$values}" size="{$p['size']}"
+                {$required} {$disabled} data-input>
+         <a class="input-button" data-toggle>
+            <i class="far fa-calendar-alt fa-lg pointer"></i>
+         </a>
+         $clear
+      </div>
+HTML;
+
+      $date_format = Toolbox::getDateFormat('js');
+
+      $min_attr = !empty($p['min'])
+         ? "minDate: '".self::convDate($p['min'])."',"
+         : "";
+      $max_attr = !empty($p['max'])
+         ? "maxDate: '".self::convDate($p['max'])."',"
+         : "";
```

Comment:
I did not tested it, but I think dates should not be converted here (see other comments).